pkg/api/handlers/compat/containers.go

```diff
@@ -620,3 +616,46 @@ func RenameContainer(w http.ResponseWriter, r *http.Request) {
 
 	utils.WriteResponse(w, http.StatusNoContent, nil)
 }
+
+// adaptNetworkSettings overcomes discrepancies between the shape of
+// NetworkSettings type used for inspect in docker and podman.
+func adaptNetworkSettings(input *define.InspectNetworkSettings) (types.NetworkSettings, error) {
+	// type alias to allow to define an annonimous field
+	type inputNetworkSettings = define.InspectNetworkSettings
+	type adaptedNetworkSettings struct {
+		// all the input fields are embedded in this annonimous field.
+		inputNetworkSettings
+
+		// these fields will shadow the correspondent embedded fields allowing us
+		// to override them.
+		SecondaryIPAddresses   []network.Address
+		SecondaryIPv6Addresses []network.Address
+	}
+
+	var adapted adaptedNetworkSettings
+	adapted.inputNetworkSettings = *input
+
+	for _, inputItem := range adapted.inputNetworkSettings.SecondaryIPAddresses {
+		adapted.SecondaryIPAddresses = append(
+			adapted.SecondaryIPAddresses,
+			network.Address{Addr: inputItem})
+	}
+
+	for _, inputItem := range adapted.inputNetworkSettings.SecondaryIPv6Addresses {
+		adapted.SecondaryIPv6Addresses = append(
+			adapted.SecondaryIPv6Addresses,
+			network.Address{Addr: inputItem})
```

Comment:
I think you also have to set the PrefixLen field in this struct, although I am not sure how we could get this value here if we do not change the type in libpod inspect as well.

Reply:
Indeed, I assumed that the strings in the current SecondaryIPv6Addresses's `[]string` in libpod inspect represent actual addresses, not CIDRs, so the default value of PrefixLen of 0 should be ok, is this correct?

Reply:
If the strings in libpod's inspect output can contain CIDRs then maybe we could try to parse them, split by `/` if present or something like that, wdyt?

Reply:
> Indeed, I assumed that the strings in the current SecondaryIPv6Addresses's `[]string` in libpod inspect represent actual addresses, not CIDRs, so the default value of PrefixLen of 0 should be ok, is this correct?

Sorry, this was wrong, if SecondaryIPv6Addresses's `[]string` in libpod inspect represent actual addresses then PrefixLen should be 64, not 0 (32 for IPv4). I've pushed changes to fix it, PTAL.

Reply:
I think this is not correct. You should have the actual ip/prefix for this container, hard coding it to 64 or 32 is wrong.
TBH looking at this I think your original patch works better. We are on 4.0 so breaking changes are possible. I do not think there many users of the secondary ip field anyway.
@mheon WDYT?

Reply:
Changing the secondary IP field in `podman inspect` itself?

You're probably right that there are almost no users. I'm not strongly opposed to making a breaking change here.